### ISSUE TITLE
Attach seqlengths to mapToTranscripts() return value

### ISSUE THREAD

**Issue by lawremi** [Bioconductor/GenomicFeatures#1]
The result of `mapToTranscripts()` treats the transcripts as the sequences. We know the lengths of the transcripts, but these are not carried over into the result. We should probably do that. I would be happy to help review this change, if desired.

**Comment by vobencha** [comment]
@dvantwisk Let's plan to work on this together next sprint.

**Comment by dvantwisk** [comment]
@lawremi, I'd like to start working on the modifications to `mapToTranscripts()` that you are suggesting but I am not entirely sure what you want to see from your description.  Could you possibly elaborate a little further and add some examples?

**Comment by lawremi** [comment]
Well I guess I'd like 

```
mapped <- pmapToTranscripts(x, tx)
identical(seqlengths(mapped), setNames(sum(width(tx)), names(tx)))
```

But for all the `*mapToTranscripts` variants. The `*mapFromTranscripts` should get a similar treatment if they don't carry over the seqlengths already.

**Comment by dvantwisk** [comment]
Here are some tests that me and @vobencha put together just to make sure that we understand what results we need for certain cases.  We would appreciate any input on whether these tests are incorrect or there or any cases that need to be considered.
```
    ## Check seqlengths() : GRanges
    x <- GRanges("A", IRanges(16, 18))
    gr <- GRanges("A", IRanges(15, 20, names="tx_b"))
    res <- mapToTranscripts(x, gr)
    checkIdentical(seqlengths(res), setNames(sum(width(res)), names(gr)))

    ## Check seqlengths() : GRangesList
    gr1 <- GRanges("A", IRanges(1, 10, names="piece_a"))
    gr2 <- GRanges("A", IRanges(15, 20, names="piece_b"))
    transcript <- GRangesList("my_transcript" = c(gr1, gr2))
    res <- mapToTranscripts(x, transcript)
    checkIdentical(seqlengths(res), setNames(sum(width(transcript)), names(transcript)))

    ## Check seqlengths() : multiple GRanges result
    gr1 <- GRanges(c(A="chr1:25-30", B="chr1:24-40"))
    gr2 <- GRanges(c(a="chr1:25-30"))
    res <- mapToTranscripts(gr2, gr1)
    checkIdentical(seqlengths(res), setNames(width(gr1), names(gr1)))

    ## Check seqlengths() : empty GRanges result
    gem <- GRanges(c(A="chr1:0-1"))
    res <- mapToTranscripts(gr2, gem)
    checkIdentical(seqlengths(res), 0L)

    ## Check seqlengths() : pmapToTranscripts GRangesList Results
    gr1 <- GRangesList(GRanges(c(A="chr1:10-20")), GRanges(c(B="chr1:20-30")), GRanges(c(C="chr1:30-40")))
    gr2 <- GRanges(c(a="chr1:10-20", b="chr1:0-1", c="chr1:30-40"))
    res <- pmapToTranscripts(gr2, gr1)
    checkIdentical(seqlengths(p), setNames(sum(width(gr1)), names(gr1)))
```

**Comment by lawremi** [comment]
I think the first test should be:
```
checkIdentical(seqlengths(res), setNames(width(gr), names(gr)))
```

Also, the seqlengths should never be affected by the results. They're totally determined by the transcripts. So the "empty GRanges result" should be:
```
 checkIdentical(seqlengths(res), setNames(width(gem), names(gem)))
```

**Comment by vobencha** [comment]
Hi @lawremi ,
We've decided to put this on the back burner. Unfortunately there are too many other projects in the pipeline of higher priority. Maybe you want to submit a pull request?

Sorry we couldn't get to this.

**Comment by lawremi** [comment]
Ok, I'd be happy to do the pull request. One issue to discuss though: pmapToTranscripts() currently returns non-mapping regions on the sequence "UNMAPPED". I find that a little surprising, because I would expect the mapped sequences to always be on the transcript sequence, even if they fall outside the bounds of the transcript (by convention starting at 1). Thoughts?

**Comment by vobencha** [comment]
I thought UNMAPPED was a decent solution for the pmap case where the ith range in 'x' does not map to the ith range in 'transcripts'. Do you have an idea for a more intuitive solution? Open to suggestions.

**Comment by lawremi** [comment]
As I said I would expect the seqname to be the name of the transcript, regardless of whether there was overlap. The basic contract of the function is that it will map to that sequence, so it should do that, but just return an empty range on it. Introducing a virtual sequence makes the contract more complicated, and it becomes impossible to implement this feature, since the virtual sequence has no real length. I guess we could define it as zero, but does it make sense to have ranges on a zero-width sequence even if they are zero-width themselves? It would just be nice to say that the names of the result are identical to the names of the transcripts, even if the range fell into the introns or outside the tx bounds.

**Comment by vobencha** [comment]
I see, so the result would be a zero-length range with the same seqname as the transcript. That does make more sense - I'm fine with that.

**Comment by lawremi** [comment]
I think this should be resolved with 15f48b64ba03541c013c77c9d71332bf08a45b02
